pkg/services/apiserver/auth/authenticator/signedinuser.go

```diff
@@ -0,0 +1,42 @@
+package authenticator
+
+import (
+	"net/http"
+	"strconv"
+
+	"k8s.io/apiserver/pkg/authentication/authenticator"
+	k8suser "k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/klog/v2"
+
+	"github.com/grafana/grafana/pkg/infra/appcontext"
+)
+
+var _ authenticator.RequestFunc = signedInUserAuthenticator
+
+func signedInUserAuthenticator(req *http.Request) (*authenticator.Response, bool, error) {
+	ctx := req.Context()
+	signedInUser, err := appcontext.User(ctx)
+	if err != nil {
+		klog.V(5).Info("failed to get signed in user", "err", err)
+		return nil, false, nil
+	}
+
+	userInfo := &k8suser.DefaultInfo{
+		Name:   signedInUser.Login,
+		UID:    signedInUser.UserUID,
+		Groups: []string{},
+		Extra:  map[string][]string{},
+	}
+
+	for _, v := range signedInUser.Teams {
+		userInfo.Groups = append(userInfo.Groups, strconv.FormatInt(v, 10))
+	}
+
+	if signedInUser.IDToken != "" {
+		userInfo.Extra["ID-Token"] = []string{signedInUser.IDToken}
```

Comment:
Hey @toddtreece, one thing: while testing this I've noticed the extra fields seems to not be sent into the extra info header, and from [this](https://pkg.go.dev/k8s.io/apiserver@v0.27.1/pkg/authentication/user#Info) I noticed that the `Extra` keys *must* be lower-case "In order to faithfully round-trip through an impersonation flow" so that might be the case I don't see the extra fields in the request on the aggregated API server?

Reply:
@ferruvich sorry about missing that one! https://github.com/grafana/grafana/pull/82893